js/backup_in.js

```diff
@@ -332,16 +354,43 @@ function backup_generate()
             return;
         }
 
+        //electrum uses different xpriv version bytes for p2wpkh-p2sh BIP49  and p2wpkh BIP84
         var xpriv = '0488ade4' + '000000000000000000' + chaincode + '00' + privatekey;
         xpriv = base58check.encode(new Buffer(xpriv, 'hex'));
-       
+
         var bip32_keychain = new BIP32(xpriv);
-        var electrum_key = bip32_keychain.derive("m/44'/0'/0'");
-        electrum_key = base58check.encode(new Buffer(electrum_key.extended_private_key, 'hex'));
+        var electrum_key_leg = bip32_keychain.derive("m/44'/0'/0'");
+        electrum_key_leg = base58check.encode(new Buffer(electrum_key_leg.extended_private_key, 'hex'));
 
         var electrum_key_ms = bip32_keychain.derive("m/100'/45'/0'");
         electrum_key_ms = base58check.encode(new Buffer(electrum_key_ms.extended_private_key, 'hex'));
-        
+
+        var electrum_key_ltc_leg = bip32_keychain.derive("m/44'/2'/0'");
+        electrum_key_ltc_leg = base58check.encode(new Buffer(electrum_key_ltc_leg.extended_private_key, 'hex'));
+
+
+        var segwit_xpriv = '049d7878' + '000000000000000000' + chaincode + '00' + privatekey;
```

Comment:
why use the "magic number" instead of the constant that you've defined in bip32.js?

Reply:
Javascript automatically converts numbers into strings if concatenated with a string. You don't need to implement any function for that. e.g. if you have a `var i = 0` and want a string, you do `var str_i = '' + i`.